src/mame/trs/agvision.cpp

```diff
@@ -0,0 +1,404 @@
+	DEVICE_INPUT_DEFAULTS( "RS232_STOPBITS", 0xff, RS232_STOPBITS_1 )
+DEVICE_INPUT_DEFAULTS_END
+
+//-------------------------------------------------
+//  machine_config
+//-------------------------------------------------
+
+void agvision_state::agvision(machine_config &config)
+{
+	this->set_clock(XTAL(14'318'181) / 16);
+
+	// basic machine hardware
+	MC6809E(config, m_maincpu, DERIVED_CLOCK(1, 1));
+	m_maincpu->set_addrmap(AS_PROGRAM, &agvision_state::agvision_mem);
+
+	PIA6821(config, m_pia_0, 0);
+	m_pia_0->readpa_handler().set(FUNC(agvision_state::pia0_pa_r));
+	m_pia_0->irqa_handler().set_inputline(m_maincpu, M6809_FIRQ_LINE);
+	m_pia_0->cb2_handler().set(FUNC(agvision_state::pia0_cb2_w));
+
+	// video hardware
+	SCREEN(config, m_screen, SCREEN_TYPE_RASTER).set_raw(3.579545_MHz_XTAL * 2, 456, 0, 320, 262, 0, 240);
+
+	MC6847_NTSC(config, m_vdg, XTAL(14'318'181) / 4); // VClk output from MC6883
+	m_vdg->set_screen(m_screen);
+	 m_vdg->hsync_wr_callback().set(m_pia_0, FUNC(pia6821_device::ca1_w));
+	m_vdg->input_callback().set(FUNC(agvision_state::sam_read));
+
+	// memory controller
+	SAM6883(config, m_sam, XTAL(14'318'181), m_maincpu);
+	m_sam->set_addrmap(2, &agvision_state::agvision_rom);			// ROM at $A000
+	 m_sam->set_addrmap(3, &agvision_state::agvision_static_ram);	// RAM at $C000
+	m_sam->set_addrmap(4, &agvision_state::agvision_io0);			//  IO at $FF00
+	m_sam->set_addrmap(5, &agvision_state::agvision_io1);			//  IO at $FF20
+	m_sam->set_addrmap(7, &agvision_state::agvision_boot);			//  IO at $FF60
+
+	RS232_PORT(config, m_rs232, default_rs232_devices, "null_modem");
+	m_rs232->set_option_device_input_defaults("null_modem", DEVICE_INPUT_DEFAULTS_NAME(ag_modem));
+
+	// internal ram
+	RAM(config, m_ram).set_default_size("16K").set_extra_options("39K,4K");;
+}
+
+//**************************************************************************
+//  ADDRESS MAPS
+//**************************************************************************
+
+void agvision_state::agvision_mem(address_map &map)
+{
+	map(0x0000, 0xffff).rw(m_sam, FUNC(sam6883_device::read), FUNC(sam6883_device::write));
+}
+
+void agvision_state::agvision_rom(address_map &map)
+{
+	// $A000-$BFFF
+	map(0x0000, 0x07ff).rom().region("maincpu", 0x0000).nopw().mirror(0x1800);
+}
+
+void agvision_state::agvision_static_ram(address_map &map)
+{
+	// $C000-$FEFF
+	map(0x0000, 0x0080).ram();
+}
+
+void agvision_state::agvision_io0(address_map &map)
+{
+	// $FF00-$FF1F
+	map(0x1c, 0x1f).rw("pia0", FUNC(pia6821_device::read), FUNC(pia6821_device::write));
+}
+
+void agvision_state::agvision_io1(address_map &map)
+{
+	// $FF20-$FF3F
+	map(0x00, 0x00).w(FUNC(agvision_state::ff20_write));
+}
+
+void agvision_state::agvision_boot(address_map &map)
+{
+	// $FF60-$FFEF
+	map(0x60, 0x7f).nopw(); // SAM Registers
+}
+
+//-------------------------------------------------
+//  device_start
+//-------------------------------------------------
+
+void agvision_state::device_start()
+{
```

Comment:
The one drivers use is machine_start(), not device_start().
And then you won't need to call the parent class function either. So, no need for driver_device::machine_start();

Reply:
Fixed.